plugins/hls-import-actions-plugin/src/Ide/Plugin/ImportActions.hs

```diff
@@ -188,47 +201,58 @@ codeActionProvider _ ideState _pId (CodeActionParams _ _ TextDocumentIdentifier
             , _disabled = Nothing
             , _data_ = data_}
 
-codeActionResolveProvider :: Recorder (WithPriority Log) -> ResolveFunction IdeState EIResolveData 'Method_CodeActionResolve
+codeActionResolveProvider :: Recorder (WithPriority Log) -> ResolveFunction IdeState IAResolveData 'Method_CodeActionResolve
 codeActionResolveProvider _ ideState _ ca _ rd =
   pluginResponse $ do
     wedit <- resolveWTextEdit ideState rd
     pure $ ca & L.edit ?~ wedit
 --------------------------------------------------------------------------------
 
-resolveWTextEdit :: IdeState -> EIResolveData -> ExceptT String (LspT Config IO) WorkspaceEdit
+resolveWTextEdit :: IdeState -> IAResolveData -> ExceptT String (LspT Config IO) WorkspaceEdit
 resolveWTextEdit ideState (ResolveOne uri int) = do
```

Comment:
TBH this `ResolveOne` name is weird. Shouldn't it be `ExplicitOne`?

Reply:
Ugh, I figured it out. We are resolving individual edits _generically_, using the same machinery for both the explicit import edits and the refine import edits.

That makes me think that we could take this faster. What if we had
```
data ResolveData = ResolveData { uri :: Uri, importEdits :: [Int] }
```
Now we can have a totally generic resolve handler: get _all_ the referenced edits, and stick them together into a workspace edit, return. It's just up to the original code action provider to select the edits it will want eventually.

Reply:
We need to know the type when we offer code actions or resolve because we have slightly different behavior between the two (titles are different for example).